packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap

```diff
@@ -3436,6 +3436,537 @@ export default function MyPostForm(props: MyPostFormProps): React.ReactElement;
+                  ariaLabel=\\"button\\"
+                  onClick={(event) => {
+                    event.stopPropagation();
+                    removeItem(index);
+                  }}
+                />
+              </Badge>
+            );
+          })}
+        </ScrollView>
+      )}
+      <Divider orientation=\\"horizontal\\" marginTop={5} />
+    </React.Fragment>
+  );
+  if (lengthLimit !== undefined && items.length >= lengthLimit && !isEditing) {
+    return arraySection;
+  }
+  return (
+    <React.Fragment>
+      <Text>{label}</Text>
+      {isEditing && children}
+      {!isEditing ? (
+        <>
+          <Button
+            onClick={() => {
+              setIsEditing(true);
+            }}
+          >
+            Add item
+          </Button>
+        </>
+      ) : (
+        <Flex justifyContent=\\"flex-end\\">
+          {(currentFieldValue || isEditing) && (
+            <Button
+              children=\\"Cancel\\"
+              type=\\"button\\"
+              size=\\"small\\"
+              onClick={() => {
+                setFieldValue(defaultFieldValue);
+                setIsEditing(false);
+                setSelectedBadgeIndex(undefined);
+              }}
+            ></Button>
+          )}
+          <Button
+            size=\\"small\\"
+            variation=\\"link\\"
+            color={tokens.colors.brand.primary[80]}
+            isDisabled={hasError}
+            onClick={addItem}
+          >
+            {selectedBadgeIndex !== undefined ? \\"Save\\" : \\"Add\\"}
+          </Button>
+        </Flex>
+      )}
+      {arraySection}
+    </React.Fragment>
+  );
+}
+export default function TagCreateForm(props) {
+  const {
+    clearOnSuccess = true,
+    onSuccess,
+    onError,
+    onSubmit,
+    onCancel,
+    onValidate,
+    onChange,
+    overrides,
+    ...rest
+  } = props;
+  const initialValues = {
+    label: undefined,
+    Posts: [],
+    statuses: [],
+  };
+  const [label, setLabel] = React.useState(initialValues.label);
+  const [Posts, setPosts] = React.useState(initialValues.Posts);
+  const [statuses, setStatuses] = React.useState(initialValues.statuses);
+  const [errors, setErrors] = React.useState({});
+  const resetStateValues = () => {
+    setLabel(initialValues.label);
+    setPosts(initialValues.Posts);
+    setCurrentPostsValue(undefined);
+    setCurrentPostsDisplayValue(\\"\\");
+    setStatuses(initialValues.statuses);
+    setCurrentStatusesValue(undefined);
+    setErrors({});
+  };
+  const [currentPostsDisplayValue, setCurrentPostsDisplayValue] =
+    React.useState(\\"\\");
+  const [currentPostsValue, setCurrentPostsValue] = React.useState(undefined);
+  const PostsRef = React.createRef();
+  const [currentStatusesValue, setCurrentStatusesValue] =
+    React.useState(undefined);
+  const statusesRef = React.createRef();
+  const postRecords = useDataStoreBinding({
+    type: \\"collection\\",
+    model: Post,
+  }).items;
+  const getDisplayValue = {
```

Comment:
What happens if there are more than one array of enums? Should we add the field name in here to make it unique?

Reply:
If there's another array of enums, it will be added as a key value pair to this object. The key is the field name and the value is the function to return the display value.